Update documentation for Datadog for IntelliJ plugin (#18642)
DataDog/documentation: content/en/developers/ide_integrations/idea/_index.md

The Datadog plugin for IntelliJ IDEA helps you to improve software performance by providing meaningful code-level insights in the IDE based on real-time observability data.

{{< img src="/developers/ide_integrations/idea/overview1.png" alt="The Datadog tool window open in IDEA" style="width:100%;" >}}

The **Code Insights** view keeps you informed about:
- issues from [Error Tracking][6]
- [vulnerability][8] reports by Application Security Management
- [flaky tests][9] detected by CI Visibility
- profiling insights from [Watchdog Insights][10]

With the **Continuous Profiler**, the plugin helps you to reduce latency and lower cloud costs by highlighting code lines that:
- consume the most CPU
- allocate the most memory
- spend the most time on locks, disk I/O, socket I/O, and so on

In addition to code highlighting, you can use the Datadog plugin for Intellij IDEA to:
- Identify methods that consume the most resources with the **Top List** view.
- Visualize aggregated profile data with the **Flame Graph** view.
- Navigate directly from the top list and flame graph to the relevant lines in your code.
- Find resource consumption broken down by method name and line number.

> The **Logs Navigation** support opens the Datadog Log Explorer with a view matching the context in which you are working.

Author: The idea is for the Overview to mention each main feature the plugin offers.  Perhaps the real issue is that the description for the Continuous Profiler is too long? (previously it was the _main_ feature)

- Double-clicking an item in the list (or selecting **Jump to Source** from the context menu) opens a source code editor showing where the method is defined.
- To see a flame graph visualization of a method, select **Search in Flame Graph** from the context menu.

#### Call tree

The call tree to the right of the **Top List** shows the paths that lead to (and from) the selected method.

The default **Caller Hierarchy** view shows the callers (or predecessors) of the target method and the frequency with which they appear in the call stack. To view the callees (or successors), click the **Callee Hierarchy** button on the toolbar.

Right-click on a method in the call tree to see options to navigate to the source editor or flame graph.

### Flame graph

> A flame graph is a visualization of profiling samples that shows stack traces and their relative frequency during the sample period. The Datadog plugin take multiple individual profiles collected over the requested time frame, and aggregates them. Each individual profile covers a 60 second interval within the requested time frame.

Reviewer: ```suggestion
A flame graph is a visualization of profiling samples that shows stack traces and their relative frequency during the sample period. The Datadog plugin collects multiple individual profiles from the requested time frame, and aggregates them. Each individual profile covers a 60 second interval within the requested time frame.
```

{{< img src="/developers/ide_integrations/idea/separate-flamegraph-by.png" alt="Use the tooltip button to separate frames by method or line number" style="width:40%;" >}}

> ### Source Highlighting

Reviewer: ```suggestion
### Source highlighting
```

> When the Continuous Profiler tab is active, the plugin adds code highlights to the source code editor margin. For Top Methods an icon appears in the editor margin, and line-level highlights appear in the code based on the active Profiling data.

Reviewer: ```suggestion
When the Continuous Profiler tab is active, the plugin adds code highlights to the source code editor margin. For Top Methods, an icon appears in the editor margin, and line-level highlights appear in the code based on the active Profiling data.
```

- Hover over the icon to see more information.
- Click the icon to open the top list Profiling tab or open Profiling in Datadog.
  {{< img src="/developers/ide_integrations/idea/interest-options.png" alt="Click the Datadog icon to open the Profiling data in a tab or in Datadog" style="width:100%;" >}}

The active Profiling tab also affects the IDEA project tree view, which is annotated with the selected profile's metrics:
{{< img src="/developers/ide_integrations/idea/project-tree-view.png" alt="The project tree annotated with profile metrics from a profile tab" style="width:60%;" >}}

> ## Logs Navigation

Reviewer: ```suggestion
## Logs navigation
```

> You can navigate to the [Log Explorer][5] on the Datadog platform directly from your Java source files. Look out for **View Logs** links following log statements in your source code:

Reviewer: ```suggestion
You can navigate to the [Log Explorer][5] on the Datadog platform directly from your Java source files. Look for the**View Logs** links following the log statements in your source code:
```

{{< img src="/developers/ide_integrations/idea/logs-navigation.png" alt="A source file showing a View Logs link." style="width:100%;" >}}

> Clicking the link opens the **Log Explorer** with a query that matches the logger name, log level and log message as closely as possible.

Reviewer: ```suggestion
Clicking the link opens the **Log Explorer** with a query that matches the logger name, log level, and log message as closely as possible.
```

> ## View In IntelliJ IDEA

Reviewer: ```suggestion
## View in IntelliJ IDEA
```

> The **View in IntelliJ** feature provides a link from the Datadog platform directly to your source files. Look out for the button next to frames in stack traces displayed on the platform (for example, in [Error Tracking][6]):

Reviewer: ```suggestion
The **View in IntelliJ** feature provides a link from the Datadog platform directly to your source files. Look for the button next to frames in stack traces displayed on the platform (for example, in [Error Tracking][6]):
```
